mlflow/store/artifact/s3_artifact_repo.py

```diff
@@ -116,9 +120,8 @@ def __init__(
         self._secret_access_key = secret_access_key
         self._session_token = session_token
 
-    def _get_s3_client(self, addressing_style="path", s3_endpoint_url=None):
+    def _get_s3_client(self):
         return _get_s3_client(
-            addressing_style=addressing_style,
```

Comment:
Removing the `addressing_style` and `s3_endpoint_url` logic here since it is only used in R2 which no longer inherits this class. Also, it seems that the `s3_endpoint_url` was not being passed to the `_get_s3_client` call originally. Should we hot fix it tomorrow?